pinot-tools/src/main/java/com/linkedin/pinot/tools/ConfigValidator.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.tools;
+
+import java.net.URL;
+import org.apache.commons.httpclient.HttpClient;
+import org.apache.commons.httpclient.methods.GetMethod;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Given controller host and controller port, validate the following configs:
+ * <ul>
+ *   <li>Table Config</li>
+ *   <li>Schema</li>
+ * </ul>
+ * <p>This validation only checks whether the config can be successfully loaded (e.g. valid JSON format).
+ * <p>No guarantee on config's content.
+ */
+public class ConfigValidator {
+  private static final Logger LOGGER = LoggerFactory.getLogger(ConfigValidator.class);
+
+  private static final HttpClient HTTP_CLIENT = new HttpClient();
+
+  private final String _controllerHost;
+  private final int _controllerPort;
+
+  public ConfigValidator(String controllerHost, int controllerPort) {
+    _controllerHost = controllerHost;
+    _controllerPort = controllerPort;
+  }
+
+  public void validateTableConfig() {
+    try {
+      URL url = new URL("http", _controllerHost, _controllerPort, "/tables");
+      GetMethod httpGet = new GetMethod(url.toString());
+      int responseCode = HTTP_CLIENT.executeMethod(httpGet);
+      String response = httpGet.getResponseBodyAsString();
+      if (responseCode != 200) {
+        LOGGER.error("While fetching table names, got error response code: {}, response: {}", responseCode, response);
+      } else {
+        JSONArray tables = new JSONObject(response).getJSONArray("tables");
+        int numTables = tables.length();
+        for (int i = 0; i < numTables; i++) {
+          String table = tables.getString(i);
+          LOGGER.info("Validating table config for table: \"{}\"", table);
+          try {
+            url = new URL("http", _controllerHost, _controllerPort, "/tables/" + table);
```

Comment:
Use a central place to build paths, instead of assuming a hard-coded path here (eg ControllerRequestBuilder, or something similar).